### ISSUE TITLE
Why does the loss of solving nonlinear ordinary differential equations begin to be so huge that it cannot be reduced even after training？

### ISSUE THREAD

**Issue by pandasayz** [lululxvi/deepxde#462]
I'd really appreciate it if someone nice could help me，I really need some help~
the nonlinear ordinary differential equations are :
d^2u/dt^2=-600*u*v-250*t
d^2v/dt^2=300*(u^2-v^2)+250*t
the initial condition are:
u(0)=0
v(0)=0
u'(0)=9.259
v'(0)=9.259
I have added a hard constraint
my code is as following:

````
import deepxde as dde
import numpy as np
import tensorflow as tf

 def ode(x, y):
     y1, y2 = y[:, 0:1], y[:, 1:2]
     y=np.vstack(y1,y2)
     dy1_xx = dde.grad.hessian(y, x, component=0, i=0, j=0)
     dy2_xx = dde.grad.hessian(y, x, component=1, i=0, j=0)
     return [
         dy1_xx +600*y1*y2+250*x ,
         dy2_xx -300*(y1**2-y2**2)-250*x
   ]

geom = dde.geometry.Interval(0, 1)

def boundary(x, on_boundary):
    return on_boundary and np.isclose(x[0], 0)


bc1 = dde.NeumannBC(geom, lambda x: np.full_like(x, 9.259),  lambda _, boundary:boundary, component=0)
bc2 = dde.NeumannBC(geom, lambda x: np.full_like(x, 9.259),  lambda _, boundary:boundary, component=1)

data = dde.data.PDE( geom, ode, [bc1, bc2],num_domain=1000, num_boundary=2)
net = dde.maps.FNN([1] + [20] * 5 + [2], "tanh", "Glorot normal")

def modify_output(x, y):
    u, v = y[:, 0:1], y[:, 1:2]  # Assume the order of output is u, v
    u_new = x * u
    v_new = x * v
    return tf.concat((u_new, v_new), axis=1)

net.apply_output_transform(modify_output)

model = dde.Model(data, net)

loss_weights=[1,1,200,200]
model.compile("adam", lr=1.0e-3,loss_weights=loss_weights)
losshistory, train_state =model.train(epochs=20000)
dde.saveplot(losshistory, train_state, issave=True, isplot=True)
````

and my results are really not good:
![问题](https://user-images.githubusercontent.com/96182400/147382244-f0a2d481-cce0-41cb-bb73-114672720f06.png)

Thank you very, very much for the good people who read my question carefully and helped me

**Comment by pandasayz** [comment]
@lululxvi I would be really appreciate it if you could take the time to look at it

**Comment by pandasayz** [comment]
@lululxvi dear Professor：Your work is really meaningful. I will really appreciate it if you could take the time to look at it

**Comment by lululxvi** [comment]
Delete `y=np.vstack(y1,y2)`. Also there are issues like scaling, check FAQ.

**Comment by pandasayz** [comment]
> Delete `y=np.vstack(y1,y2)`. Also there are issues like scaling, check FAQ.

Thank you very much for taking the time out of your busy schedule to reply me. I have already tried to delete y=np.vstack(y1,y2), but unfortunatly The results still don't look good：

![问题1](https://user-images.githubusercontent.com/96182400/147384645-d4be73c4-7cee-43de-ae8a-0d76261c4758.png)

I am very sorry that I did not see this content in your article about DeepXde. Do you have any examples of such treatment? I'm not quite sure which part of the program to change and so on? Really thank you again.

**Comment by lululxvi** [comment]
Here is the FAQ: https://deepxde.readthedocs.io/en/latest/user/faq.html

**Comment by pandasayz** [comment]
> Here is the FAQ: https://deepxde.readthedocs.io/en/latest/user/faq.html

Thank you again for your help~Sorry to bother you again：
 I read some of the questions carefully, and I found that in my equations，for example:
d^2u/d^2t+600*u*v+250*t=0
the d^2u/d^2t∈[-1e-2,1e-2]
and 600*u*v∈[-1,-1e-9]
and 250*t∈[0,1e2]

what I want to ask is : How do I deal with the difference in the magnitude of the terms in the equation
Thank you again for your help！

**Comment by pandasayz** [comment]
what I do not understand is: Should I look at the magnitude of each items ? or should I just look at the magnitude of the whole thing on the equation's left-hand side?
thank you very much!

**Comment by pandasayz** [comment]
I find when t∈[0,1e-3],the solution seems good:
![11](https://user-images.githubusercontent.com/96182400/147623995-f0d97af6-5bb2-473a-9b49-31a0cbd8ddb0.png)


but if I want know the solution when t∈[1e-3,1], what should I do ?
I really need your help，Thank you again for your help！

**Comment by lululxvi** [comment]
There are two ways for scaling: (1) change of variable, (2) have the scaling in network. There are examples in FAQ.

Resolving the scaling should be helpful for longer time.

**Comment by pandasayz** [comment]
> There are two ways for scaling: (1) change of variable, (2) have the scaling in network. There are examples in FAQ.
> 
> Resolving the scaling should be helpful for longer time.

Thank you very much for your help. I will check it carefully. Best wishes!

**Comment by MINE0126** [comment]
Hello, @pandasayz 
Regarding the question: "How to deal with differences in scale of terms in an equation?" Have you made progress accordingly?
Do you have any good suggestions to help me with a similar scale difference problem? Thank you very much!
